crates/evm-prover/src/prover/prover.rs

```diff
@@ -0,0 +1,3 @@
+//! TODO: Types for block prover will live here
+//!
+//! pub struct BlockProver {}
```

Comment:
The plan is to put a `BlockProver` struct here which will encapsulate interactions with sp1 ProverClient and rpcs for data inputs. Basically refactor out a lot of the blevm stuff